### PR TITLE
Adding sequenceM method to Traverse

### PR DESCRIPTION
**Pull request by zainab-ali**

### CODE REVIEW

File: core/src/main/scala/cats/Traverse.scala [hunk 1]
```diff
@@ -84,6 +84,24 @@ import simulacrum.typeclass
...
+   * scala> import cats.implicits._
+   * scala> val x: List[Option[List[Int]]] = List(Some(List(1, 2)), Some(List(3)))
+   * scala> val y: List[Option[List[Int]]] = List(None, Some(List(3)))
+   * scala> x.sequenceM
+   * res0: Option[List[Int]] = Some(List(1, 2, 3))
+   * scala> y.sequenceM
+   * res1: Option[List[Int]] = None
+   * }}}
+   */
+  def sequenceM[G[_], A](fgfa: F[G[F[A]]])(implicit G: Applicative[G], F: FlatMap[F]): G[F[A]] =
```
Comment (adelbertc) [inline core/src/main/scala/cats/Traverse.scala]: sequenceA? since applicative

Comment (kailuowang) [inline core/src/main/scala/cats/Traverse.scala]: Well this one is corresponding `traverseM`, maybe both should be renamed? (if we ended up doing that we need to add the "breaking change" label. )

Comment (zainab-ali) [inline core/src/main/scala/cats/Traverse.scala]: What about `flatTraverse` or `flatSequence`, taking the convention from Scala's `flatMap`?  I realize that this isn't well established in other languages, but both `traverseM` and `sequenceM` aren't as self explanatory

Comment (peterneyens) [inline core/src/main/scala/cats/Traverse.scala]: There was a similar conversation when I added `traverseM` in #1020.

Comment (kailuowang) [inline core/src/main/scala/cats/Traverse.scala]: Sorry for the delayed response. I like `flatTraverse` and `flatSequence`, they make sense to me.

**Comment by kailuowang** [comment]
Thanks very much! @zainab-ali 👍  LGTM.

**Comment by travisbrown** [comment]
:+1:

**Comment by peterneyens** [comment]
I was not convinced about the name change at first, but after playing around with it I actually like the analogy between `flatMap` and `flatTraverse` : 

```scala
def smallerNaturals(i: Int): List[Int] = 
  List.range(1, Math.max(i, 1))

def smallerNaturalsOpt(i: Int): Option[List[Int]] = 
  if (i <= 1) None else Some(List.range(1, i))

List(2,3,4).flatMap(smallerNaturals)
// List[Int] = List(1, 1, 2, 1, 2, 3)

List(2,3,4).flatTraverse(smallerNaturalsOpt)
// Option[List[Int]] = Some(List(1, 1, 2, 1, 2, 3))
```

So :+1: , thanks @zainab-ali.

**Pull request merged**